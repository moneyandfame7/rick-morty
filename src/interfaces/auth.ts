export interface AuthCredentials {
  email: string;
  password: string;
}

export interface Role {
  id: number;
  value: string;
}

export interface User {
  readonly id: string;
  readonly email: string;
  readonly username: string | null;
  readonly auth_type: string;
  readonly banned: boolean;
  readonly role: Role;
  readonly country: string | null;
  readonly photo: string | null;
  readonly mail_subscribe: boolean | null;
}

export interface AuthResponse {
  refresh_token: string;
  access_token: string;
  user: User;
}

export interface Token {
  id: number;
  user_id: string;
  refresh_token: string;
}
