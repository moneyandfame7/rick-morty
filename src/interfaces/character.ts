export interface ICharacter {
  id: number
  created: string
  episode: Array<string>
  gender: string
  image: string
  location: {
    name: string
    url: string
  }
  name: string
  url: string
  origin: {
    name: string
    url: string
  }
  species: string
  status: string
}
