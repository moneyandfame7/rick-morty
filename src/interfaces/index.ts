interface IEntity {
  id: number;
  name: string;
  createdAt: Date;
}

interface IResponse {
  info: {
    page: number;
    take: number;
    count: number;
    pages: number;
    next: string | null;
    prev: string | null;
  };
}

export type CharacterGender = "Female" | "Male" | "Genderless" | "unknown";
export type CharacterStatus = "Alive" | "Dead" | "unknown";
export type CharacterLocation = Omit<ILocation, "residents">;

export interface ICharacter {
  id: number;
  status: CharacterGender;
  name: string;
  species: string;
  gender: CharacterStatus;
  type: string;
  image: string;
  createdAt: Date;
  origin: CharacterLocation;
  location: CharacterLocation;
  episodes: number[];
}

export interface ILocation {
  id: number;
  name: string;
  type: string;
  dimension: string;
  createdAt: Date;
  residents: number[];
}

export interface IEpisode extends IEntity {
  air_date: string;
  episode: string;
  characters: string[];
  url: string;
}

export interface IResponseCharacter extends IResponse {
  results: ICharacter[];
}

export interface IResponseEpisode extends IResponse {
  results: IEpisode[];
}

export interface IResponseLocation extends IResponse {
  results: ILocation[];
}
