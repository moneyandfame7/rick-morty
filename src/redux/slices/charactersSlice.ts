import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import { FAVORITE_CHARACTERS, getLocalStorage } from "../../utils/localStorage";
import { ICharacter } from "../../interfaces";
import { concat, filter } from "lodash";
interface CharacterState {
  characters: ICharacter[];
}

const initialState: CharacterState = {
  characters: getLocalStorage(FAVORITE_CHARACTERS),
};

export const charactersSlice = createSlice({
  name: "characters",
  initialState,
  reducers: {
    addToFavorite: (state, action: PayloadAction<ICharacter>) => {
      state.characters = concat(state.characters, action.payload);
    },
    removeFromFavorite: (state, action: PayloadAction<number>) => {
      if (state.characters) {
        state.characters = filter(state.characters, c => c.id !== action.payload);
      }
    },
  },
});

export const { addToFavorite, removeFromFavorite } = charactersSlice.actions;

export default charactersSlice.reducer;
