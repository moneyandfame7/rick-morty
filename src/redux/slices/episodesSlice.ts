import { IEpisode } from "../../interfaces";
import { AnyAction, createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
import { makeConcurrentRequest } from "../../utils/fetch";
interface IEpisodeState {
  fetchedEpisodes: IEpisode[];
  loading: boolean;
  error: string | null;
}
const initialState: IEpisodeState = {
  fetchedEpisodes: [],
  loading: false,
  error: null,
};
export const fetchEpisodes = createAsyncThunk<IEpisode[], string[], { rejectValue: string }>(
  "episodes/fetchEpisodes",
  async (episodes: any, { rejectWithValue }) => {
    const response = await makeConcurrentRequest(episodes);

    if (response) {
      return response;
    } else {
      return rejectWithValue("Failed to fetch episodes");
    }
  }
);
export const episodesSlice = createSlice({
  name: "episodes",
  initialState,
  reducers: {},
  extraReducers: builder => {
    builder
      .addCase(fetchEpisodes.pending, state => {
        state.loading = true;
        state.error = null;
      })
      .addCase(fetchEpisodes.fulfilled, (state, action: PayloadAction<IEpisode[]>) => {
        state.loading = false;
        state.fetchedEpisodes = action.payload;
      })
      .addMatcher(isError, (state, action: PayloadAction<string>) => {
        state.loading = false;
        state.error = action.payload;
      });
  },
});

function isError(action: AnyAction) {
  return action.type.endsWith("rejected");
}

export default episodesSlice.reducer;
