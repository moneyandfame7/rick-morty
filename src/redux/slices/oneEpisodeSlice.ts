import { AnyAction, createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
import { IEpisode } from "../../interfaces";
import { getApiResource } from "../../utils/fetch";
import { isError, isLoading } from "../selectors";

interface IEpisodeState {
  oneEpisode: string;
  loading: boolean;
  error: string | null;
}
const initialState: IEpisodeState = {
  oneEpisode: "",
  loading: false,
  error: null,
};

export const fetchEpisodeByUrl = createAsyncThunk<string, string, { rejectValue: string }>(
  "oneEpisode/fetch",
  async (url: string, { rejectWithValue }) => {
    const response = await getApiResource<IEpisode>(url);

    if (response) {
      return response.name;
    } else {
      return rejectWithValue("Failed to fetch episodes");
    }
  }
);

export const oneEpisodeSlice = createSlice({
  name: "oneEpisode",
  initialState,
  reducers: {},
  extraReducers: builder => {
    builder
      .addCase(fetchEpisodeByUrl.fulfilled, (state, action: PayloadAction<string>) => {
        state.loading = false;
        state.oneEpisode = action.payload;
      })
      .addMatcher(isLoading, state => {
        state.loading = true;
        state.error = null;
      })
      .addMatcher(isError, (state, action: PayloadAction<string>) => {
        state.loading = false;
        state.error = action.payload;
      });
  },
});

export default oneEpisodeSlice.reducer;
