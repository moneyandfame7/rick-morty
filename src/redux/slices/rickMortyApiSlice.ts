import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { ICharacter, IEpisode, IResponseCharacter, IResponseEpisode } from "../../interfaces";
import { BASE_URL, NavigationTypeEnum } from "../../constants/api";

export const apiSlice = createApi({
  reducerPath: "api",
  baseQuery: fetchBaseQuery({
    baseUrl: BASE_URL,
  }),
  endpoints: builder => ({
    fetchCharacters: builder.query<IResponseCharacter, number | void>({
      query: (page = 1) => `${NavigationTypeEnum.CHARACTER}/?page=${page}`,
    }),
    fetchCharacterById: builder.query<ICharacter, number | void>({
      query: id => `/${NavigationTypeEnum.CHARACTER}/${id}`,
    }),
    fetchEpisodes: builder.query<IResponseEpisode, number | void>({
      query: page => `/${NavigationTypeEnum.EPISODE}/?page=${page}`,
    }),
    fetchEpisodeById: builder.query<IEpisode, number | void>({
      query: id => `/${NavigationTypeEnum.EPISODE}/${id}`,
    }),
  }),
});

export const { useFetchCharactersQuery, useFetchCharacterByIdQuery, useFetchEpisodesQuery, useFetchEpisodeByIdQuery } =
  apiSlice;
