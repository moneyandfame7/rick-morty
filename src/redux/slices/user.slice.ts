import { User } from "../../interfaces/auth";
import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { getLocalStorage, setLocalStorage } from "../../utils/localStorage";
import { LOCALSTORAGE_USER } from "../../constants/api";

interface AuthState {
  user: User | null;
}

const initialState: AuthState = {
  user: getLocalStorage(LOCALSTORAGE_USER),
};

export const userSlice = createSlice({
  name: "credentials",
  initialState,
  reducers: {
    setUser: (state, action: PayloadAction<User>) => {
      setLocalStorage(LOCALSTORAGE_USER, action.payload);
      state.user = action.payload;
    },
    removeUser: state => {
      state.user = null;
    },
  },
});

export const { setUser, removeUser } = userSlice.actions;

export default userSlice.reducer;
