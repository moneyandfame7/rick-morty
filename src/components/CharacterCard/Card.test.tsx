import { render, screen } from "@testing-library/react";
import { CharacterCard } from "./index";
import { useFetchEpisodeByIdQuery } from "../../redux/slices/rickMortyApiSlice";
import { IEpisode } from "../../interfaces";
import React from "react";

jest.mock("../../redux/slices/rickMortyApiSlice", () => ({
  useFetchEpisodeByIdQuery: jest.fn(),
}));

jest.mock("../../utils/getIdFromUrl", () => ({
  getIdFromName: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
}));

const data = {
  status: "Alive",
  name: "Test Name",
  image: "",
  location: {
    name: "location test",
    url: "url test",
  },
  episode: ["episode test1", "episode test2", "episode test3"],
  id: 1,
};

describe("CharacterCard", () => {
  let component: any;

  const mockEpisode = {
    name: "Test Name",
    episode: "Test Episode",
  };

  it("should render properly with default", () => {
    givenEpisode();
    givenComponent();
    thenItRendersProperly();
  });

  it("should render properly with Dead status", () => {
    givenEpisode();
    givenComponent({ status: "Dead" });
    thenItRendersProperly();
  });

  function givenComponent(props?: any) {
    component = render(<CharacterCard {...data} {...props} />);
  }

  function givenEpisode(data: Partial<IEpisode> = mockEpisode, isLoading: boolean = false) {
    (useFetchEpisodeByIdQuery as jest.Mock).mockImplementation(() => ({
      isLoading,
      data,
    }));
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }
});
