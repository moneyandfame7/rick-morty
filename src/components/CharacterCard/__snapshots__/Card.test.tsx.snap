// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should render properly with Dead status 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            width="200px"
          />
          <div
            class="MuiCardHeader-root css-13fwgw0-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiCardHeader-title css-1l8qytf-MuiTypography-root"
              >
                Test Name
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiCardHeader-subheader css-zzcnnq-MuiTypography-root"
              >
                location test
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-71xg00-MuiCardContent-root"
          >
            <div
              class="MuiBox-root css-uwdvlb"
            >
              <h4
                class="MuiTypography-root MuiTypography-h5 css-wr0xd3-MuiTypography-root"
              >
                Dead
              </h4>
            </div>
            <div
              class="MuiBox-root css-1vr2anf"
            >
              <h5
                class="MuiTypography-root MuiTypography-subtitle1 css-1smqr0y-MuiTypography-root"
              >
                First seen in:
              </h5>
              <div
                class="css-1ouuv2g-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
                >
                  Test Name
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
                >
                  (Test Episode)
                </h6>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="green iguana"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          width="200px"
        />
        <div
          class="MuiCardHeader-root css-13fwgw0-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiCardHeader-title css-1l8qytf-MuiTypography-root"
            >
              Test Name
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiCardHeader-subheader css-zzcnnq-MuiTypography-root"
            >
              location test
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-71xg00-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-uwdvlb"
          >
            <h4
              class="MuiTypography-root MuiTypography-h5 css-wr0xd3-MuiTypography-root"
            >
              Dead
            </h4>
          </div>
          <div
            class="MuiBox-root css-1vr2anf"
          >
            <h5
              class="MuiTypography-root MuiTypography-subtitle1 css-1smqr0y-MuiTypography-root"
            >
              First seen in:
            </h5>
            <div
              class="css-1ouuv2g-MuiStack-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
              >
                Test Name
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
              >
                (Test Episode)
              </h6>
            </div>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterCard should render properly with default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            width="200px"
          />
          <div
            class="MuiCardHeader-root css-13fwgw0-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiCardHeader-title css-1l8qytf-MuiTypography-root"
              >
                Test Name
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiCardHeader-subheader css-zzcnnq-MuiTypography-root"
              >
                location test
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-71xg00-MuiCardContent-root"
          >
            <div
              class="MuiBox-root css-uwdvlb"
            >
              <h4
                class="MuiTypography-root MuiTypography-h5 css-1p5zgwg-MuiTypography-root"
              >
                Alive
              </h4>
            </div>
            <div
              class="MuiBox-root css-1vr2anf"
            >
              <h5
                class="MuiTypography-root MuiTypography-subtitle1 css-1smqr0y-MuiTypography-root"
              >
                First seen in:
              </h5>
              <div
                class="css-1ouuv2g-MuiStack-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
                >
                  Test Name
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
                >
                  (Test Episode)
                </h6>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-zwt9xa-MuiPaper-root-MuiCard-root"
      >
        <img
          alt="green iguana"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          width="200px"
        />
        <div
          class="MuiCardHeader-root css-13fwgw0-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiCardHeader-title css-1l8qytf-MuiTypography-root"
            >
              Test Name
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiCardHeader-subheader css-zzcnnq-MuiTypography-root"
            >
              location test
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-71xg00-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-uwdvlb"
          >
            <h4
              class="MuiTypography-root MuiTypography-h5 css-1p5zgwg-MuiTypography-root"
            >
              Alive
            </h4>
          </div>
          <div
            class="MuiBox-root css-1vr2anf"
          >
            <h5
              class="MuiTypography-root MuiTypography-subtitle1 css-1smqr0y-MuiTypography-root"
            >
              First seen in:
            </h5>
            <div
              class="css-1ouuv2g-MuiStack-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
              >
                Test Name
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-516ssh-MuiTypography-root"
              >
                (Test Episode)
              </h6>
            </div>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
